Resources/public/js/views/services/ez-roleserversideviewservice.js

```diff
@@ -0,0 +1,300 @@
+                // TODO UDW should be limited to the Users subtree, and return User/UserGroup objects: EZP-24917
+                userIdStr = usersInfoObject._href + '/' + content.get('contentId');
+
+                userService.assignRoleToUser(
+                    userIdStr, roleAssignInputStruct, callback
+                );
+            });
+        },
+
+        /**
+         * Assign the role to a user group
+         *
+         * @method _assignRoleToUserGroup
+         * @protected
+         * @param {Object} role
+         * @param {Object|null} limitation
+         * @param {Content} content
+         * @param {Function} callback the callback function
+         */
+        _assignRoleToUserGroup: function (role, limitation, content, callback) {
+            var userService = this.get('capi').getUserService(),
+                discoveryService = this.get('capi').getDiscoveryService(),
+                roleAssignInputStruct= userService.newRoleAssignInputStruct(role, limitation),
+                location,
+                groupIdStr;
+
+            discoveryService.getInfoObject('rootUserGroup', function (error, rootUserGroup) {
+                if (error) {
+                    callback(error);
+                    return;
+                }
+
+                /* We need to convert content ID to user group ID, e.g.
+                 change this: /api/ezp/v2/content/locations/1/5/14
+                 into this: /api/ezp/v2/user/groups/1/5/14
+                 */
+                // TODO UDW should be limited to the Users subtree, and return User/UserGroup objects: EZP-24917
+                location = content.get('resources').MainLocation;
+                groupIdStr = /.+user\/groups/.exec(rootUserGroup._href) + /(\/\d+)+$/g.exec(location)[0];
+
+                userService.assignRoleToUserGroup(
+                    groupIdStr, roleAssignInputStruct, callback
+                );
+            });
+        },
+
+        /**
+         * Assign role callback triggering notification and calling given callback
+         *
+         * @method _assignRoleCallback
+         * @protected
+         * @param {String} roleId the role id
+         * @param {String} roleName the role name
+         * @param {Array} contents the array of Content items to which role is being assigned to
+         * @param {Integer} countAssigned number of successfully assignments
+         * @param {Function} callback the callback to call when other tasks are done
+         */
+        _assignRoleCallback: function (roleId, roleName, contents, countAssigned, callback) {
+            var notificationIdentifier = this._getAssignRoleNotificationIdentifier(
+                    'assign-role', roleId, contents
+                );
+
+            if (countAssigned>0) {
+                this._notify(
+                    '"' + roleName + '" role has been assigned to ' + countAssigned + ' Content items',
```

Comment:
For these messages I have been using "users/groups" rathern than "content items" since you cannot assign a role to an article, for example. A bit clearer, I think. Same goes for the other messages.